### PR TITLE
Split annotation into substring. Url encode them. Call urls.

### PR DESCRIPTION
**Pull request by simoneduca**

### CODE REVIEW

**Comment by simoneduca** [comment]
@rogerhutchings I think this is ready to go, if you wouldn't mind having a look at it.

File: app/modules/transcribe/annotations/annotations.factory.js [hunk 1]
```diff
@@ -57,8 +60,36 @@ function AnnotationsFactory(localStorageService) {
         return annotation;
     }
 
+
+    function checkVariants(annotation) {
+        //splits string by whitespace (different encodings)
+        var lemmas = annotation.text.split(/\s|&nbsp;/g);
```
Comment (rogerhutchings) [inline app/modules/transcribe/annotations/annotations.factory.js]: Why `\s` and `&nbsp;`? Also, are you stripping out any punctuation? I'm guessing there are places where periods have actual meaning, but commas etc?

File: app/modules/transcribe/annotations/annotations.factory.js [hunk 2]
```diff
@@ -57,8 +60,36 @@ function AnnotationsFactory(localStorageService) {
...
+        var results = [];
```
Comment (rogerhutchings) [inline app/modules/transcribe/annotations/annotations.factory.js]: Why are you keeping `results` here? It's not available outside this function, and it's not added to the annotation or anything that I can see..?

File: app/modules/transcribe/modals/variants.controller.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    var vm = this;
+    vm.close = close;
+    vm.data = AnnotationsFactory.list();
+    vm.variants = getVariants;
+
+    function close(result) {
+        $modalInstance.close(result);
+    }
+
+    function getVariants() {
```
Comment (rogerhutchings) [inline app/modules/transcribe/modals/variants.controller.js]: Remove empty function?

File: app/modules/transcribe/transcribe.controller.js [hunk 4]
```diff
@@ -45,6 +42,23 @@ function TranscribeController($stateParams, $modal, $scope, $window, Annotations
...
 
+    function variantsFeedback() {
+        var annotations = AnnotationsFactory.list();
+        var vArray = []
+        annotations.forEach(function (el) {
+            if (el.type === 'text' && el.variants) {
+                vArray.push(el.variants);
+            }
+        });
+        if (vArray) {
```
Comment (rogerhutchings) [inline app/modules/transcribe/transcribe.controller.js]: This is always going to be true, as an empty array is a truthy value in Javascript :trollface: Use `(vArray.length)` instead, as `0` is always falsy.

File: app/modules/transcribe/annotations/annotations.factory.js [hunk 5]
```diff
@@ -57,8 +60,36 @@ function AnnotationsFactory(localStorageService) {
...
+            results.push(urlLemmas);
+            $http({
+                method: 'GET',
+                url: 'https://static.zooniverse.org/www.shakespearesworld.org/variants/' + results[i] + '.txt'
+            }).then(function successCallback(response) {}, function errorCallback(response) {
+                var url = response.config.url;
+                var words = decodeURIComponent(url.substring(url.lastIndexOf("/") + 1, url.lastIndexOf(".")));
+                oedVars.push(words);
+                upsert(_.extend(annotation, {
+                    variants: oedVars
```
Comment (rogerhutchings) [inline app/modules/transcribe/annotations/annotations.factory.js]: Okay, this is a bit strange... assuming `oedVars` should be different for each lemma, how come you're adding the previous variants to the annotation for each new one? Shouldn't you create a new `oedVars` array for each lemma?

File: app/modules/transcribe/annotations/annotations.factory.js (hunk 1 above)
Comment (simoneduca) [inline app/modules/transcribe/annotations/annotations.factory.js]: `\s` match all white spaces and `&nbsp;` is added when sw-tags are added. So I need them to have the words separated correctly. Good call on commas, I'll add them in.

File: app/modules/transcribe/annotations/annotations.factory.js (hunk 2 above)
Comment (simoneduca) [inline app/modules/transcribe/annotations/annotations.factory.js]: Do you mean it should be inside the loop?

Comment (rogerhutchings) [inline app/modules/transcribe/annotations/annotations.factory.js]: I don't know what it does, it's not returned or used anywhere else. Should it be there at all?

File: app/modules/transcribe/annotations/annotations.factory.js (hunk 5 above)
Comment (simoneduca) [inline app/modules/transcribe/annotations/annotations.factory.js]: Not sure I follow. `oedVars` contains the possible variants in any give annotation. Can you give me an example of what you mean in code please?

File: app/modules/transcribe/transcribe.controller.js (hunk 4 above)
Comment (simoneduca) [inline app/modules/transcribe/transcribe.controller.js]: Good call, thanks!

File: app/modules/transcribe/modals/variants.controller.js (hunk 3 above)
Comment (simoneduca) [inline app/modules/transcribe/modals/variants.controller.js]: Yes I forgot in there when I was trying to display the actual possible variants in the html.

File: app/modules/transcribe/annotations/annotations.factory.js (hunk 5 above)
Comment (rogerhutchings) [inline app/modules/transcribe/annotations/annotations.factory.js]: Oh right. If it's not meant to be the variants for a single lemma, then fine

File: app/modules/transcribe/annotations/annotations.factory.js (hunk 2 above)
Comment (simoneduca) [inline app/modules/transcribe/annotations/annotations.factory.js]: Got it. I'll get rid of it.

**Pull request merged**